dotsql.go

```diff
@@ -58,21 +59,37 @@ type ExecerContext interface {
 
 // DotSql represents a dotSQL queries holder.
 type DotSql struct {
-	queries map[string]string
+	queries map[string]*template.Template
+	data    any
 }
 
-func (d DotSql) lookupQuery(name string) (query string, err error) {
-	query, ok := d.queries[name]
+func (d DotSql) WithData(data any) DotSql {
+	return DotSql{queries: d.queries, data: data}
+}
+
+func (d DotSql) lookupQuery(name string, data any) (query string, err error) {
+	template, ok := d.queries[name]
 	if !ok {
 		err = fmt.Errorf("dotsql: '%s' could not be found", name)
 	}
+	if template != nil {
+		buffer := bytes.NewBufferString("")
+		if err != nil {
+			return
+		}
+		err = template.Execute(buffer, data)
```

Comment:
I'm not familiar with `Execute`, in which cases it can fail?

Reply:
`text/template` is pretty extensive with what you can pass in, including objects and functions.

If you passed in a function that panicked. Or if you used some logic like `{{ if .user.phone }}` and `.user` was nil, those would certainly fail at execution time.